core/services/chainlink/config.go

```diff
@@ -0,0 +1,43 @@
+package chainlink
+
+import (
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/chains/solana"
+	"github.com/smartcontractkit/chainlink/core/chains/terra"
+	"github.com/smartcontractkit/chainlink/core/config/toml"
+)
+
+// Config is TODO doc
+//
+// When adding a new field:
+// 	- consider including a unit suffix with the field name
+// 	- TOML is limited to int64/float64, so fields requiring greating range/precision must use non-standard types
+//	implementing encoding.TextMarshaler/TextUnmarshaler
+type Config struct {
+	toml.CoreConfig
+
+	EVM []EVMConfig `toml:",omitempty"`
+
+	Solana []SolanaConfig `toml:",omitempty"`
+
+	Terra []TerraConfig `toml:",omitempty"`
+}
+
+type EVMConfig struct {
+	ChainID int64 `toml:",omitempty"` //TODO big.Int?
```

Comment:
Why? Just in case? We already limit to `int64` internally in some places FWIW, but the tradeoff here is that switching to `big.Int` requires a UX change from number to string in the TOML:
```diff
-ChainID = 10012
+ChainID = '10012'
```

Reply:
Done